R/ensure-lib.R

```diff
@@ -86,6 +90,31 @@ ensure_lib <- function(lib = NULL, test_packages = unlist(strsplit(packageDescri
   lib_dir_path
 }
 
+install_arrow_github <- function(repo = "apache/arrow", ref = "HEAD", ...) {
+  # TODO: check if cmake is found? Say to use brew to install that?
+  # Do we want or need to also allow this to install any github package. For
+  # most it would only need to change the repo arg + expose the subdir argument
+  with_envvar(
+    list(
+      # always use the tools/linuxlibs.R script for installing arrow lib
+      USE_TOOLS_LIBS_SCRIPT = "true",
```

Comment:
This requires an arrow change right? We can 🚲 🏠  the name there.

Reply:
yeah, I went through a few others and this seems the least super-hacky but not super happy about it lemme send that as a PR and we can decide there before we merge this I'll update